inst/include/dust/distr/binomial.hpp

```diff
@@ -23,22 +23,26 @@ double binomial_inversion(double n, double prob, RNG& generator) {
   return num_geom;
 }
 
-inline double stirling_approx_tail(double k) {
-  static double kTailValues[] = {0.0810614667953272,  0.0413406959554092,
+template <typename real_t>
+real_t stirling_approx_tail(real_t k) {
+  static real_t kTailValues[] = {0.0810614667953272,  0.0413406959554092,
                                  0.0276779256849983,  0.02079067210376509,
                                  0.0166446911898211,  0.0138761288230707,
                                  0.0118967099458917,  0.0104112652619720,
                                  0.00925546218271273, 0.00833056343336287};
   if (k <= 9) {
-    return kTailValues[static_cast<int>(k)];
+    return kTailValues[static_cast<short>(k)];
```

Comment:
I hadn't come across `short` before. Usually I use `uint16_t` as I think it's very clear how much space is being used. This is probably moot though, I'm not even sure why this case is needed here

Reply:
This reverts back to `int` now leaving the tf code largely untouched